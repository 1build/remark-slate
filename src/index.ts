import plugin, { transform, defaultNodeTypes } from './serializer';

export { transform, defaultNodeTypes };

export default plugin;
