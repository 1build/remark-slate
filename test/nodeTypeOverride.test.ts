import { transform } from '../src';

it('transforms heading depth one to slate state object with custom node types & {type: "h1", children: [{text: "hey"}]}', () => {
  expect(
    transform(
      { type: 'heading', depth: 1, children: [{ value: 'hey' }] },
      { nodeTypes: { heading: { 1: 'h1' } } }
    )
  ).toMatchSnapshot();
});

it('Transform heading depth two without changing the default nodeTypes while passing other overrides', () => {
  expect(
    transform(
      { type: 'heading', depth: 2, children: [{ value: 'yo' }] },
      { nodeTypes: { heading: { 1: 'h1' } } }
    )
  ).toMatchSnapshot();
});

it('Transform a paragraph and assign the new nodeType', () => {
  expect(
    transform(
      { type: 'paragraph', children: [{ value: 'random text' }] },
      { nodeTypes: { paragraph: 'p' } }
    )
  ).toMatchSnapshot();
});
