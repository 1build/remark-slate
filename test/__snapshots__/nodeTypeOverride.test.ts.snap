// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transform a paragraph and assign the new nodeType 1`] = `
Object {
  "children": Array [
    Object {
      "text": "random text",
    },
  ],
  "type": "p",
}
`;

exports[`Transform heading depth two without changing the default nodeTypes while passing other overrides 1`] = `
Object {
  "children": Array [
    Object {
      "text": "yo",
    },
  ],
  "type": "heading_two",
}
`;

exports[`transforms heading depth one to slate state object with custom node types & {type: "h1", children: [{text: "hey"}]} 1`] = `
Object {
  "children": Array [
    Object {
      "text": "hey",
    },
  ],
  "type": "h1",
}
`;
